netmaster/k8snetwork/networkpolicy.go

```diff
@@ -1,36 +1,114 @@
 package networkpolicy
 
 import (
-	"encoding/json"
 	"fmt"
-	"reflect"
-	"strings"
-	"time"
-
 	log "github.com/Sirupsen/logrus"
 	"github.com/contiv/netplugin/contivmodel/client"
 	"github.com/contiv/netplugin/utils/k8sutils"
-	"k8s.io/api/networking/v1"
+	v1 "k8s.io/api/core/v1"
+	network_v1 "k8s.io/api/networking/v1"
 	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/client-go/kubernetes"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
 )
 
 const defaultTenantName = "default"
-const defaultNetworkName = "net"
+const defaultNetworkName = "default-net"
 const defaultSubnet = "10.1.2.0/24"
-const defaultEpgName = "ingress-group"
+
+//const defaultEpgName = "ingress-group"
+//const defaultEpgName = "default-epg"
+const defaultEpgName = "default-group"
 const defaultPolicyName = "ingress-policy"
 const defaultRuleID = "1"
-
+const defaultPolicyPriority = 2
+
+type k8sPodSelector struct {
+	TenantName  string //Attach Tenant
+	NetworkName string //Attach network
+	GroupName   string //Attach EPG
+	PolicyName  string //Attach to policy
+	labelPodMap map[string]map[string]bool
+	podIps      map[string]string
+}
+type k8sPolicyPorts struct {
+	Port     int
+	Protocol string
+}
+type k8sNameSelector struct {
+	nameSpaceSel string
+}
+type podCache struct {
+	labeSelector string
+	podNetwork   string
+	podGroup     string
+	podEpg       string
+}
+type k8sIPBlockSelector struct {
+	subNetIps []string
+}
+type k8sIngress struct {
+	IngressRules           []k8sPolicyPorts
+	IngressPodSelector     []*k8sPodSelector
+	IngressNameSelector    *k8sNameSelector
+	IngressIpBlockSelector *k8sIPBlockSelector
+}
+type npPodInfo struct {
+	nameSpace      string
+	labelSelectors []string
+	IP             string //??? should care for ipv6 address
+}
+type k8sNetworkPolicy struct {
+	PodSelector *k8sPodSelector
+	Ingress     []k8sIngress
```

Comment:
I think the policy name should be here instead of in the pod selector for instance